tools/querysets.py

```diff
@@ -0,0 +1,32 @@
+from django.db.models import Q, QuerySet
+
+
+class UserToolQuerySet(QuerySet):
+    def for_user(self, user):
+        return self.filter(user=user)
+
+    def visible_to_user(self, user, own_tools=True):
+        """Filter to the UserTools a user is allowed to view"""
+
+        # user is on clearance list for tools marked as public or cleared
+        cleared_tools = Q(
+            permissions__cleared_user=user,
+            clearance=self.model.Clearance.cleared.value,
+            visibility__in=[
+                self.model.Visibility.public.value,
```

Comment:
is public needed here?

Reply:
I think my reasoning was to make sure the intersection of things that are set visible that you have clearance to aren't filtered out. I would like to get this tested when we get testing setup.